developer/src/kmcmplib/tests/gtest-compiler-test.cpp

```diff
@@ -401,6 +402,76 @@ TEST_F(CompilerTest, GetXStringImpl_type_single_quote_test) {
     // CERR_StringInVirtualKeySection *** TODO ***
 }
 
+// tests strings starting with 'i'
+TEST_F(CompilerTest, GetXStringImpl_type_i_test) {
+    KMX_WCHAR tstr[128];
+    fileKeyboard.version = VERSION_80;
+    KMX_WCHAR str[LINESIZE];
+    KMX_WCHAR output[GLOBAL_BUFSIZE];
+    PKMX_WCHAR newp = nullptr;
+    PFILE_STORE file_store = new FILE_STORE[100];
+    fileKeyboard.cxStoreArray = 3u;
+    fileKeyboard.dpStoreArray = file_store;
+    file_store[1].fIsStore = TRUE;
+    u16cpy(file_store[0].szName, u"a");
+    u16cpy(file_store[1].szName, u"b");
+    u16cpy(file_store[2].szName, u"c");
+
+    // index, CERR_InvalidInVirtualKeySection *** TODO ***
+
+    // index, no close delimiter => NULL
+    u16cpy(str, u"index(");
+    EXPECT_EQ(CERR_InvalidIndex, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+
+    // index, empty delimiters => empty string
+    u16cpy(str, u"index()");
+    EXPECT_EQ(CERR_InvalidIndex, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+
+    // index, no comma or space
+    u16cpy(str, u"index(b)");
+    EXPECT_EQ(CERR_InvalidIndex, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+
+    // index, CERR_StoreDoesNotExist
+    u16cpy(str, u"index(d 4)");
+    EXPECT_EQ(CERR_StoreDoesNotExist, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+
+    // index, comma, valid
+    fileKeyboard.cxStoreArray = 3u;
+    fileKeyboard.dpStoreArray = file_store;
+    u16cpy(str, u"index(b,4)");
+    EXPECT_EQ(CERR_None, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+    const KMX_WCHAR tstr_index_comma_valid[] = { UC_SENTINEL, CODE_INDEX, 2, 4, 0 };
+    EXPECT_EQ(0, u16cmp(tstr_index_comma_valid, tstr));
+
+    // index, space, valid
+    u16cpy(str, u"index(b 4)");
+    fileKeyboard.cxStoreArray = 3u;
+    fileKeyboard.dpStoreArray = file_store;
+    EXPECT_EQ(CERR_None, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+    const KMX_WCHAR tstr_index_space_valid[] = { UC_SENTINEL, CODE_INDEX, 2, 4, 0 };
+    EXPECT_EQ(0, u16cmp(tstr_index_space_valid, tstr));
+
+    // index, two-digit parameter, valid
+    u16cpy(str, u"index(b,42)");
+    fileKeyboard.cxStoreArray = 3u;
+    fileKeyboard.dpStoreArray = file_store;
+    EXPECT_EQ(CERR_None, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+    const KMX_WCHAR tstr_index_two_digit_valid[] = { UC_SENTINEL, CODE_INDEX, 2, 42, 0 };
+    EXPECT_EQ(0, u16cmp(tstr_index_two_digit_valid, tstr));
+
+    // index, comma, non-digit parameter, CERR_InvalidIndex
+    u16cpy(str, u"index(b,g)");
+    fileKeyboard.cxStoreArray = 3u;
+    fileKeyboard.dpStoreArray = file_store;
+    EXPECT_EQ(CERR_InvalidIndex, GetXStringImpl(tstr, &fileKeyboard, str, u"", output, 80, 0, &newp, FALSE));
+
+    // index, comma, no parameter, CERR_InvalidIndex
+    u16cpy(str, u"index(b,)");
```

Comment:
Add whitespace versions, e.g. `index( b, )`, `index( b ,)`?

Reply:
`index( b,4)`, `index(b, 4)`, `index(b ,)` and `index(b, )` added